core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -713,19 +713,14 @@ protected QB copyQuery(QB query) throws IOException {
      * @return a new {@link QueryShardContext} based on the base test index and queryParserService
      */
     protected static QueryShardContext createShardContext() {
-        QueryShardContext queryCreationContext = new QueryShardContext(queryShardContext);
-        queryCreationContext.reset();
-        queryCreationContext.parseFieldMatcher(ParseFieldMatcher.STRICT);
-        return queryCreationContext;
+        return new QueryShardContext(queryShardContext);
```

Comment:
This is a shard context, so these parse field matcher should always be backed by indexSettings. If we want to default to STRICT here we should change that in the setup of the test I think.